src/pretix/base/email.py

```diff
@@ -258,9 +258,30 @@ def _placeholder_payment(order, payment):
         return str(payment.payment_provider.order_pending_mail_render(order))
 
 
+def get_best_name(position_or_address, parts=False):
+    """
+    Return the best name we got for either an invoice address or an order position, falling back to the respective other
+    """
+    from pretix.base.models import InvoiceAddress, OrderPosition
+    if isinstance(position_or_address, InvoiceAddress):
+        if position_or_address.name:
+            return position_or_address.name_parts if parts else position_or_address.name
+        position_or_address = position_or_address.order.positions.exclude(attendee_name_cached="").exclude(attendee_name_cached__isnull=True).first()
```

Comment:
what if such a position does not exist?

Reply:
Then it won't be an instance of OrderPosition (since None is not an instance of anything) and we'll fall back to the empty string/dict, since we haven't got anything